lib/matplotlib/dviread.py

```diff
@@ -808,14 +812,14 @@ class PsfontsMap(object):
     """
     __slots__ = ('_font', '_filename')
 
-    def __init__(self, filename):
+    @lru_cache()
+    def __new__(cls, filename):
```

Comment:
This seems a bit dodgy. Would a helper function (still using lru_cache) also work here instead of decorating `__new__`?